CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7)
+cmake_minimum_required(VERSION 3.7...3.30)
```

Comment:
Why is this change important/relevant?

Reply:
It's for CMake 4.0 compatibility. So not directly related, could open another PR for this if desired.